### ISSUE TITLE
Search broken for certain queries with certain buckets

### ISSUE THREAD

**Issue by drewbitt** [winpax/sfsu#36]
I can reproduce this in 2 different environments, both on 1.1.4. Searching is normally working fine except certain queries. E.g.

```
(base) PS C:\Users\drewbitt> scoop search ms
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Error("expected value", line: 8, column: 3)', src/bin/search.rs:47:87
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

```
(base) PS C:\Users\andys-pc> scoop search m
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Error("invalid type: integer `20140603`, expected a string", line: 17, column: 20)', src/bin/search.rs:47:87
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Error("control character (\\u0000-\\u001F) found while parsing a string", line: 11, column: 18)', src/bin/search.rs:47:87
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Error("key must be a string", line: 9, column: 2)', src/bin/search.rs:47:87
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Error("invalid type: integer `11`, expected a string", line: 17, column: 14)', src/bin/search.rs:47:87
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Error("expected `,` or `}`", line: 24, column: 3)', src/bin/search.rs:47:87
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Error("expected `,` or `}`", line: 25, column: 3)', src/bin/search.rs:47:87
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Error("expected `,` or `}`", line: 25, column: 3)', src/bin/search.rs:47:87
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Error("control character (\\u0000-\\u001F) found while parsing a string", line: 15, column: 18)', src/bin/search.rs:47:87
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Error("invalid type: integer `120707`, expected a string", line: 16, column: 18)', src/bin/search.rs:47:87
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Error("key must be a string", line: 75, column: 42)', src/bin/search.rs:47:87
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Error("key must be a string", line: 13, column: 50)', src/bin/search.rs:47:87
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Error("key must be a string", line: 69, column: 42)', src/bin/search.rs:47:87
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Error("control character (\\u0000-\\u001F) found while parsing a string", line: 11, column: 17)', src/bin/search.rs:47:87
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Error("expected value", line: 8, column: 3)', src/bin/search.rs:47:87
```

Other examples, `scoop search zn` works but `scoop search ag` doesn't. Longer searches that don't return results like `scoop search znqsad1qw4e` works.

Env:
Powershell 7, Windows 10, using the profile addition to powershell
```
(base) PS C:\Users\andys-pc> scoop bucket list

Name                   Source                                        Updated               Manifests
----                   ------                                        -------               ---------
anderlli0053_DEV-tools https://github.com/anderlli0053/DEV-tools     12/15/2022 8:53:39 PM     10988
dorado                 https://github.com/chawyehsu/dorado           12/15/2022 7:11:32 PM       227
extras                 https://github.com/ScoopInstaller/Extras      12/15/2022 7:33:23 PM      1745
java                   https://github.com/ScoopInstaller/Java        12/15/2022 8:12:53 PM       253
main                   https://github.com/ScoopInstaller/Main        12/15/2022 7:34:12 PM      1126
nonportable            https://github.com/ScoopInstaller/Nonportable 12/15/2022 8:48:52 PM       104
```

**Comment by drewbitt** [comment]
Removing the `anderlli0053_DEV-tools` repository fixes the problem. But sfsu shouldn't be crashing here. If there's something wrong with the repository it should be handled

**Comment by jewlexx** [comment]
TLDR: Most of the panics come from invalid manifests, for which I am happy to add an "invalid manifest" label, but I will not add specific support for it.

From what I can tell the manfiests that are causing the panic have multi-line post_install scripts, control characters, comments, etc.

`dcmo5.json` breaks because it has an integer version number, where there should be a string. This does not follow the scoop spec.
Many others break because they have control characters where they shouldn't, which breaks the JSON spec.
Some others have comments, which again breaks the JSON spec.

I will not be putting time and effort into supporting manifests that do not obide by the specs which are provided by scoop and ECMA respectively.

I will however, add special cases for when the parsing panics, which can show that the manifests were invalid, but will not provide detailed information that may be provided for valid manifests.
